### PR TITLE
[CIS-383] Remove .none Filter option

### PR DESCRIPTION
**Pull request by mantergo**
**In this PR:**

- Remove `.none` `Filter` option
- Adjust `NewUserQueryUpdater` to handle `nil` filter
- Adjust `Sample` app after introducing `nil` `Filter`

### CODE REVIEW

File: Sources_v3/Query/UserListQuery.swift [hunk 1]
```diff
@@ -25,24 +25,29 @@ public struct UserListQuery: Encodable {
...
     public func encode(to encoder: Encoder) throws {
         var container = encoder.container(keyedBy: CodingKeys.self)
-        try container.encode(filter, forKey: .filter)
-        
+        switch filter {
+        case let .some(value): try container.encode(value, forKey: .filter)
+        case .none:
+            // Backend expects empty object for "filter_conditions" in case no filter specified.
+            struct EmptyObject: Encodable {}
+            try container.encode(EmptyObject(), forKey: .filter)
```
Comment (mantergo) [inline Sources_v3/Query/UserListQuery.swift:48]: I don't like this part 🙂
Do you know more elegant way to simulate empty object?
Backend refuses to accept `null` or empty string.

Comment (VojtaStavik) [reply Sources_v3/Query/UserListQuery.swift:48]: Yeah, this is unfortunately needed. You can do something like:
```swift
try container.encode([String](), forKey: .filter)
```
But to honest, I find you solution easier to understand and I don't mind having it in the codebase.

File: Sources_v3/Database/DTOs/UserDTO.swift [hunk 2]
```diff
@@ -92,6 +92,16 @@ extension NSManagedObjectContext: UserDatabaseSession {
         
         return dto
     }
+    
+    func updateQuery(
+        for userId: UserId,
+        queryFilterHash: String
+    ) throws {
```
Comment (VojtaStavik) [inline Sources_v3/Database/DTOs/UserDTO.swift:99]: Either is the `throws` here redundant or you forgot to throw some errors :)

File: Sources_v3/Workers/Background/NewUserQueryUpdater_Tests.swift [hunk 3]
```diff
@@ -57,11 +57,29 @@ class NewUserQueryUpdater_Tests: StressTestCase {
...
+        try database.createUser()
+        
+        // Assert `update(userListQuery` is not called for .none filter query
+        AssertAsync.willBeTrue(env!.userQueryUpdater?.update_queries.isEmpty)
+    }
+    
+    func test_newUser_updated_withNilFilterQuery() throws {
+        let id: UserId = .unique
+        
+        try database.createUserListQuery(filter: .none)
```
Comment (VojtaStavik) [inline Sources_v3/Workers/Background/NewUserQueryUpdater_Tests.swift:76]: This `.none` is `Optional.none` not `Filter.none` 🤯 It took me a while to understand. However, I'd rather use `nil` :D :D

File: Sample_v3/Samples/CombineSimpleChat/CombineSimpleChannelsViewController.swift [hunk 4]
```diff
@@ -236,8 +236,7 @@ class CombineSimpleChannelsViewController: UITableViewController {
         else { return }
         
         usersViewController.userListController = chatClient
-            // TODO: Change filter to all users after Filter adjustments
-            .userListController(query: .init(filter: .autocomplete("name", with: "a")))
+            .userListController(query: .init())
```
Comment (VojtaStavik) [inline Sample_v3/Samples/CombineSimpleChat/CombineSimpleChannelsViewController.swift:239]: Maybe `.init()` should be the default parameter of `. userListController()` then?

File: Sources_v3/Workers/Background/NewUserQueryUpdater_Tests.swift (hunk 3 above)
Comment (mantergo) [reply Sources_v3/Workers/Background/NewUserQueryUpdater_Tests.swift:76]: Haha, I changed it to `nil` in another test, but forgot here cause it doesn't throw compilation error 😅

File: Sample_v3/Samples/CombineSimpleChat/CombineSimpleChannelsViewController.swift (hunk 4 above)
Comment (mantergo) [reply Sample_v3/Samples/CombineSimpleChat/CombineSimpleChannelsViewController.swift:239]: I thought about it, but probably we want developers always use query explicitly, even if it default one.

File: Sources_v3/Query/UserListQuery.swift [hunk 5]
```diff
@@ -25,24 +25,29 @@ public struct UserListQuery: Encodable {
...
         self.filter = filter
         self.sort = sort
         self.pagination = pagination
     }
     
     public func encode(to encoder: Encoder) throws {
         var container = encoder.container(keyedBy: CodingKeys.self)
-        try container.encode(filter, forKey: .filter)
-        
+        switch filter {
```
[the remark this answers is not in the document]
Comment (mantergo) [reply Sources_v3/Query/UserListQuery.swift:43]: Nope 👀

**Pull request merged**